man/systemctl.xml

```diff
@@ -1690,111 +1690,114 @@ Jan 12 10:46:45 example.com bluetoothd[8900]: gatt-time-server: Input/output err
+            asynchronous behavior.</para>
           </listitem>
         </varlistentry>
 
         <varlistentry>
           <term><command>rescue</command></term>
 
           <listitem>
-            <para>Enter rescue mode. This is mostly equivalent to
-            <command>isolate rescue.target</command>, but also prints a
-            wall message to all users.</para>
+            <para>Enter rescue mode. This is mostly equivalent to <command>systemctl isolate rescue.target</command>,
+            but also prints a wall message to all users. This operation is blocking by default, use
+            <option>--no-block</option> to request asynchronous behavior.</para>
           </listitem>
         </varlistentry>
         <varlistentry>
           <term><command>emergency</command></term>
 
           <listitem>
-            <para>Enter emergency mode. This is mostly equivalent to
-            <command>isolate emergency.target</command>, but also prints
-            a wall message to all users.</para>
+            <para>Enter emergency mode. This is mostly equivalent to <command>systemctl isolate
+            emergency.target</command>, but also prints a wall message to all users. This operation is blocking by
+            default, use <option>--no-block</option> to request asynchronous behavior.</para>
           </listitem>
         </varlistentry>
         <varlistentry>
           <term><command>halt</command></term>
 
           <listitem>
-            <para>Shut down and halt the system. This is mostly equivalent to <command>start halt.target
+            <para>Shut down and halt the system. This is mostly equivalent to <command>systemctl start halt.target
             --job-mode=replace-irreversibly</command>, but also prints a wall message to all users.  If combined with
             <option>--force</option>, shutdown of all running services is skipped, however all processes are killed and
             all file systems are unmounted or mounted read-only, immediately followed by the system halt.  If
             <option>--force</option> is specified twice, the operation is immediately executed without terminating any
             processes or unmounting any file systems. This may result in data loss. Note that when
-            <option>--force</option> is specified twice the halt operation is executed by
-            <command>systemctl</command> itself, and the system manager is not contacted. This means the command should
-            succeed even when the system manager hangs or crashed.</para>
+            <option>--force</option> is specified twice the halt operation is executed by <command>systemctl</command>
+            itself, and the system manager is not contacted. This means the command should succeed even when the system
+            manager hangs or crashed. This operation is blocking by default, use <option>--no-block</option> to request
```

Comment:
You made a parenthetical comment that the sysvinit commands didn't block, which would also apply here...

If the queuing succeeds - i.e. the asynchronous version would have returned success - we've almost certainly scheduled to kill the `systemctl` right away.  Leaving `halt` and `kexec` as different only affects cases like the debug shell.  Why not favour UI simpliclity here?  What do we gain by exposing the ability to halt synchronously, in the dedicated command?  (when no-one uses `halt` anyway, because we restored the historical meaning, leaving their hardware converting electricity to lights and humming noises).

sidenote: if halt/kexec don't use logind, then they don't send wall messages either.